services/brig/src/Brig/API/User.hs

```diff
@@ -246,9 +247,12 @@ createUser new@NewUser {..} = do
       Nothing -> throwE InvalidInvitationCode
     ensureMemberCanJoin tid = do
       maxSize <- fromIntegral . setMaxTeamSize <$> view settings
-      mems <- lift $ Intra.getTeamMembers tid
-      when (length (mems ^. Team.teamMembers) >= maxSize) $
-        throwE TooManyTeamMembers
+      case Range.checked maxSize of
+        Nothing -> throwE TooManyTeamMembers
```

Comment:
This means that we can't increase `setMaxTeamSize` beyond the 2000 limit, which is not what we want.  The 2000 limit is on the truncated list of *some* users of a team; the `setMaxTeamSize` config value may be bigger.

Unless I'm missing something?

Reply:
I was thinking for Large Teams, we will probably want to branch behaviour here. So, when we eventually have a way we will write some code here to not fail. But for now, nobody should be able to create a team larger than 2000.
LMK if my assumption is wrong.